core/src/main/java/com/redhat/lightblue/client/request/data/GenerateRequest.java

```diff
@@ -0,0 +1,98 @@
+package com.redhat.lightblue.client.request.data;
+
+
+import java.util.List;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.redhat.lightblue.client.Operation;
+import com.redhat.lightblue.client.Projection;
+import com.redhat.lightblue.client.Query;
+import com.redhat.lightblue.client.Sort;
+import com.redhat.lightblue.client.http.HttpMethod;
+import com.redhat.lightblue.client.request.AbstractLightblueDataRequest;
+
+/**
+ * Generate a value using a field's value generator
+ *
+ * path: Path for the field whose value generator will be invoked
+ * n: Number of values to generate, defaults to 1
+ *
+ * Parse the results as:
+ * <pre>
+ *	String[] results = client.data(request, String[].class);
+ * </pre>
+ * 
+ */
+public class GenerateRequest extends AbstractLightblueDataRequest {
+
+    private Integer n;
+    private String path;
```

Comment:
Given null could be passed to constructor it's not going to fix anything.  If anything have getRestURI fail if there's no path?  Would save a request to lightblue if it was malformed.